packages/blocks/src/generators/arduino.ts

```diff
@@ -376,6 +395,28 @@ export class Arduino extends Blockly.Generator {
 		return `setup${sensorName}();\n`;
 	}
 
+	public getI2CChannel(block: Block) {
+		let current = block.getSurroundParent();
+		while (current) {
+			if (current.type === "i2c_use_channel") {
+				return Number.parseInt(current.getFieldValue("CHANNEL"));
+			}
+			current = current.getSurroundParent();
+		}
+
+		return null;
+	}
+
+	public addI2CDevice(prefix: string, block: Block, component: Component) {
```

Comment:
To make it a bit more clear this function is for the schema's:
```suggestion
	public addI2CDeviceToSchema(prefix: string, block: Block, component: Component) {
```
And some comments/jsdoc might be nice as well, esp in a common class like this.